polus-k-means-clustering-plugin/src/main.py

```diff
@@ -0,0 +1,244 @@
+from sklearn.metrics import davies_bouldin_score 
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(logging.INFO)
+
+def list_file(csv_directory):
+    """List all the .csv files in the directory.
+    
+    Args:
+        csv_directory (str): Path to the directory containing the csv files.
+        
+    Returns:
+        The path to directory, list of names of the subdirectories in dirpath (if any) and the filenames of .csv files.
+        
+    """
+    list_of_files = [os.path.join(dirpath, file_name)
+                     for dirpath, dirnames, files in os.walk(csv_directory)
+                     for file_name in fnmatch.filter(files, '*.csv')]
+    return list_of_files
+   
+def elbow(data, minimumrange, maximumrange):
+    """Determine k value using elbow method.
+    
+    Args:
+        data(dataframe): Input csvfile.
+        minimumrange(int): Starting number of sequence in range function to determine k-value.
+        maximumrange(int):Ending number of sequence in range function to determine k-value.
+    Returns:
+        Optimal k-value.
+    
+    """
+    data_array = np.array(data)
+    sse = []
+    K = range(minimumrange, maximumrange)
+    for k in K:
+        kmeans = KMeans(n_clusters = k).fit(data_array)
+        centroids = kmeans.cluster_centers_
+        pred_clusters = kmeans.predict(data_array)
+        curr_sse = 0
+        # calculate square of Euclidean distance of each point from its cluster center
+        for i in range(len(data_array)):
+            curr_center = centroids[pred_clusters[i]]
+            curr_sse += (data_array[i, 0] - curr_center[0]) ** 2 + (data_array[i, 1] - curr_center[1]) ** 2
+        sse.append(curr_sse)
+    #Find the elbow point in the curve
+    points = len(sse)
+    coord = np.vstack((range(points), sse)).T
+    np.array([range(points), sse])
+    f_point = coord[0]
+    linevec = coord[-1] - coord[0]
+    linevecn = linevec / np.sqrt(np.sum(linevec**2))
+    vecf = coord - f_point
+    prod = np.sum(vecf * np.matlib.repmat(linevecn, points, 1), axis=1)
+    vecfpara = np.outer(prod, linevecn)
+    vecline= vecf - vecfpara
+    dist = np.sqrt(np.sum(vecline ** 2, axis=1))
+    k_cluster = np.argmax(dist)
+    return k_cluster
+    
+def calinski_harabasz(data, minimumrange, maximumrange):
+    """Determine k value using Calinski Harabasz Index method.
+    
+    Args:
+        data(dataframe): Input csvfile.
+        minimumrange(int): Starting number of sequence in range function to determine k-value.
+        maximumrange(int):Ending number of sequence in range function to determine k-value.
+        
+    Returns:
+        Optimal k-value.
+    
+    """
+    data_array = np.array(data)
+    if minimumrange <=1:
+        raise ValueError('Minimumrange should be greater than 1.')
```

Comment:
You do not perform the same check in the `elbow` method. I highly recommend removing this code here and putting it in the main section of code. By putting all common code outside of individual functions, you remove the likelihood of having these kinds of issues where you do error checks in some methods but not others.